File: python/easy/find_duplicates_in_range.py
def find_duplicates_in_range(arr,l,r,target):
  counter = 0
  start = 0
  end = 0
# O(n)

  if l < 0:
    start = 0
  else:
    start = l

  if r < 0:
    end = 0
  else:
    end = r

  if start > end:
    [start, end] = [end, start]

  for item in arr[start:end]:
    if item == target:
      counter += 1
  return counter

print(find_duplicates_in_range([0,0,0,0,1,2,3,4,5,6,7,8,9,9,9,0,10], 1, 5, 0))
print(find_duplicates_in_range([0,0,0,0,1,2,3,4,5,6,7,8,9,9,9,0,10], 1, 5, 9))
print(find_duplicates_in_range([0,0,0,0,1,2,3,4,5,6,7,8,9,9,9,0,10], 1, -1, 0))
# O(n)

File: js/easy/findDuplicatesInRange.js
function findDuplicatesInRange(arr, l, r, target) {
  var counter = 0;
  var start = l < 0 ? 0 : l;
  var end = r < 0 ? 0 : r;

  if (start > end) {
    [start, end] = [end, start];
  }

  for (var i = l; i < r; i++) {
    if (arr[i] === target) {
      counter++;
    }
  }
  return counter;
}
console.log(findDuplicatesInRange([0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 0], 1, 5, 0));
console.log(
  findDuplicatesInRange(
    [0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 0, 9, 9, 9, 10],
    1,
    5,
    9,
  ),
);
// O(n)
